File: solutions/scala/anagram/1/src/main/scala/Anagram.scala
object Anagram {
  def isAnagram(base:String, cand:String) = {
    val baseL = base.toLowerCase
    val candL = cand.toLowerCase
    baseL != candL && baseL.sorted == candL.sorted
  }

  def findAnagrams(str:String, candidates:List[String]):List[String] = {
    candidates filter(x => isAnagram(str, x))
  }
}

File: solutions/erlang/anagram/1/src/anagram.erl
-module(anagram).

-export([find_anagrams/2]).

is_anagram(Subject, Candidate) ->
    string:to_lower(Subject) =/= string:to_lower(Candidate) andalso
        lists:sort(string:to_lower(Subject)) == lists:sort(string:to_lower(Candidate)).

find_anagrams(Subject, Candidates) ->
    lists:filter(fun(Candidate) -> is_anagram(Subject, Candidate) end, Candidates).